### DISCUSSION TITLE
'dagger listen' without arguments is unusable

### DISCUSSION THREAD

**Discussion by shykes** [dagger/dagger#12277] [Help]
## Problem

`dagger listen` generates a token by default. But the token is not printed to the user, so it's impossible to connect.
The only valid way to call `dagger listen` to day is by explicitly passing the token as an argument.

## Solution

Print the full URL (with token) so that `dagger listen` is usable without arguments.

**Comment by pjmagee** [comment]
Is this my problem... the CLI output doesnt mention a --token. I've been trying to make dotnet test run by invoking a session in the test runner not having to wrap in dagger run dotnet test 

and I've been doing dagger listen --listen 127.0.0.1:{CUSTOM_PORT} but didnt know you can pass a token?

**Comment by helderco** [comment]
I use `dagger listen` a lot, but I do it like this:

```bash
DAGGER_SESSION_TOKEN=test dagger listen --allow-cors
```

Then I use this to connect: [https://studio.apollographql.com/sandbox/explorer](https://studio.apollographql.com/sandbox/explorer) (endpoint [http://127.0.0.1:8080/query](http://127.0.0.1:8080/query), authorization header `Basic dGVzdDo=` as the base64 encoded `test` token).

Usually the env var is already in my env. I do agree that it needs to work with the flags alone, but I'll probably want to keep using it like this to keep the token the same, so I don't have to update the credentials in Apollo Studio.
